### DISCUSSION TITLE
notification generated uuid

### DISCUSSION THREAD

**Discussion by hamada-emam-tech** [laravel/framework#48304] [Q&A] [answered]
hello, i'm in use of Database notifications with oracle database, i'm deal with views and procedures i need to customize DatabaseNotification class, it still generating an id for every record i need to make database do this not the notification system, i traced the generation of id it is in sendNow function into NotificationSender class it generate notificationId like                 $notificationId = Str::uuid()->toString(); in line 102 then set the notification id to id in the function sendToNotifiable with this code      
   if (! $notification->id) {
            $notification->id = $id;
        }
all i need is to stop this behavior, is there a way to do so ?

**Comment by chuckrincon** [comment, marked as answer]
Hey  ✋🏻

You can't stop it because it's too deep in the core to override the default behavior.
Instead, you could listen to the `Illuminate\Notifications\Events\NotificationSent` event and update the notification with your custom `ID`.


`EventServiceProvider`
```php
use App\Listeners\UpdateNotification;
use Illuminate\Notifications\Events\NotificationSent;
 
/**
 * The event listener mappings for the application.
 *
 * @var array
 */
protected $listen = [
    NotificationSent::class => [
        UpdateNotification::class,
    ],
];
```

`UpdateNotification`
```php
/**
 * Handle the event.
 */
public function handle(NotificationSent $event): void
{
    $event->notification->update(['id' => Model::getCustomKey()])
}
```

With this approach, when a notification is dispatched by the notification system, the listener will update the `id` generated by the core, with your custom `id`

**Reply by hamada-emam-tech** [reply]
Thanks, ...

**Comment by pedroni** [comment]
The solution gave by @chuckrincon is almost correct, I tried implementing like mentioned but it didn't work. Instead you need to check for `response` on the `NotificationSent` event instance, if it's a model, then update it.

To be honest, I'd prefer that laravel generated `orderedUuid` by default, since ordering by `created_at` is really slow when multiple notifications are being sent (which was the reason why I ended up here)

```php

namespace App\Listeners;

use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Notifications\Events\NotificationSent;
use Illuminate\Support\Str;

class NotificationSentListener implements ShouldQueue
{
    public function handle(NotificationSent $event)
    {
        if ($event->response instanceof Model) {
            $event->response->update(['id' => Str::orderedUuid()->toString()]);
        }
    }
}

**Reply by hamada-emam-tech** [reply]
Thanks, now I'm using laravel default uuid like you mentioned.
